db/controllers/api.controller.js

```diff
@@ -53,3 +54,30 @@ exports.getComments = (req, res) => {
       res.status(500).send({ msg: "Internal Server Error" });
     });
 };
+
+exports.postComment = (req, res) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+
+  if (!username || !body) {
+    return res.status(400).send({ msg: "Bad request" });
+  }
+
+  selectArticleById(article_id)
+    .then(() => {
+      addComment(username, body, article_id)
+        .then((newComment) => {
+          res.status(201).send({ comment: newComment[0] });
+        })
+        .catch((err) => {
+          res.status(500).send({ msg: "Internal Server Error" });
+        });
+    })
+    .catch(() => {
+      if (isNaN(article_id)) {
+        res.status(400).send({ msg: "Article ID should be a number" });
+      } else {
+        res.status(404).send({ msg: "Article not found" });
+      }
+    });
```

Comment:
There is no need for multiple catch blocks also a catch block should be passing errors to the error handling middleware with `next`. Please review [the notes](l2c.northcoders.com/courses/sd-notes/back-end/#sectionId=error-handling,step=error-handling-middleware) on error handling